api/src/main/java/org/openmrs/Role.java

```diff
@@ -25,20 +38,51 @@
  *
  * @see Privilege
  */
-public class Role extends BaseChangeableOpenmrsMetadata {
+
+@Entity
+@Table(name="role")
+public class Role extends BaseOpenmrsObject {
 	
 	public static final long serialVersionUID = 1234233L;
 	
 	private static final Logger log = LoggerFactory.getLogger(Role.class);
 	
-	// Fields
-	
+	@Id
+	@Column(name = "role")
 	private String role;
 	
+	@Transient
+	@Field
+	private String name;
```

Comment:
So I'm still kind of curious why we need this?

Reply:
so this was to make sure that the [name field](https://github.com/openmrs/openmrs-core/blob/df241b5827c76661d1aa769ca39d7c8d8ee06f35/api/src/main/java/org/openmrs/BaseOpenmrsMetadata.java#L35-L37C22) doesnot get persisted in the role table where we only require role,uuid and description.

Reply:
Right, but Role now extends `BaseOpenmrsObject` which doesn't have that field any more right?

Reply:
(I do think this should implement the `OpenmrsMetadata` interface, just with `getName()` and `setName()` modifying the role).

Reply:
> Right, but Role now extends `BaseOpenmrsObject` which doesn't have that field any more right?

oh yeah, my bad




> (I do think this should implement the `OpenmrsMetadata` interface, just with `getName()` and `setName()` modifying the role).

remember this? -> https://talk.openmrs.org/t/switching-from-xml-mappings-to-annotations/39277/4?u=mherman22

Reply:
Well, sure. What I actually care about is ensuring Role implements the `Auditable` and `Retireable` interfaces.

Reply:
awesome, so what i gather here is that role should implement `OpenmrsMetadata` interface and that this ticket -> https://openmrs.atlassian.net/browse/RESTWS-910 is not required